src/controller/CHIPDeviceController.cpp

```diff
@@ -1141,11 +1141,10 @@ void DeviceCommissioner::RendezvousCleanup(CHIP_ERROR status)
         // for IP commissioning, we have taken a reference to the
         // operational node to send the completion command.
         ReleaseCommissioneeDevice(mDeviceInPASEEstablishment);
-
-        if (mPairingDelegate != nullptr)
-        {
-            mPairingDelegate->OnPairingComplete(status);
-        }
+    }
+    if (CHIP_NO_ERROR != status && mPairingDelegate != nullptr)
```

Comment:
The actual behavior here is very complicated (e.g. if your pairing code is _right_ but the device is not advertising that's the same situation as wrong pairing code!), and worse yet existing clients depend on it, so probably cannot be changed without breaking lots of shipping software.

The right answer here is that we should have a sane error reporting API, not what we have right now, but we should be _extremely_ careful about any changes to the existing behavior.